### PR TITLE
Radical update of mesh_gl plugin

### PR DESCRIPTION
**Pull request by invor**
Removes call/chaining concept for GPU data (meshes, materials, render tasks) in favor of monolithic mesh rendering modules. Still makes use of the existing collection classes for GPU data internally. New base classes are added for mesh renderers to provide a framework for implementing new mesh renderers with as little effort as possible.

## Summary of Changes

- Remove calls for GPU mesh, material and render task data
- Remove abstract base classes for GPU mesh, material and render task modules with chaining
- Add new base classes for monolithic rendering modules 
- Add modules for rendering gltf files with deferred shading and obj files with simple forward shading

### CODE REVIEW

File: plugins/mesh_gl/include/mesh_gl/MeshCalls_gl.h [hunk 1]
```diff
@@ -7,71 +7,10 @@
```
Comment (moritz-h) [inline plugins/mesh_gl/include/mesh_gl/MeshCalls_gl.h]: Why not just delete that file?

**Review by moritz-h — CHANGES_REQUESTED** [review]

File: plugins/mesh_gl/include/mesh_gl/MeshCalls_gl.h (hunk 1 above)
Comment (invor) [reply plugins/mesh_gl/include/mesh_gl/MeshCalls_gl.h]: Happy?

**Review by moritz-h — DISMISSED** [review]

**Review by moritz-h — APPROVED** [review]

**Pull request merged**